Containerfile

```diff
@@ -1,4 +1,4 @@
-FROM quay.io/forem/ruby:3.0.2 as builder
+FROM quay.io/forem/ruby:3.0.2-testing as builder
```

Comment:
This is a temporary change to confirm buildkite works when given a compatible image.

I did build these locally (/scripts/build_containers.sh) and did not observe any errors.

We should revert this base image tag after the branch is merged.

I believe that's the needed sequence, update to a temporary tag here, merge the node 16 branch into forem/forem:main, merge the infrastructure branch to promote the fedora 35 image to primary (giving it the existing tag `ruby:3.0.2`) and then reverting this change (using `FROM ruby:3.0.2 as builder` which will be the same base image).

Reply:
hmm... buildkite still doesn't work with this setup - but it might be because the docker pull commands in the build_containers script are incompatible? I'll continue with local testing of this in the meantime but buildkite fails with an error early when calling dnf to update/install packages:

```

Step 1/20 : FROM quay.io/forem/ruby:3.0.2-testing as builder
3.0.2-testing: Pulling from forem/ruby
4545346f2a49: Pull complete
838a9b65ba57: Pull complete
Digest: sha256:522930f5dd280c0b7aba5c9eb0ba19f3fed53a59822ddceed5d303ea64c08c86
Status: Downloaded newer image for quay.io/forem/ruby:3.0.2-testing
 ---> 99593d0a8366
Step 2/20 : USER root
 ---> Running in d9ebaee0052b
Removing intermediate container d9ebaee0052b
 ---> e9f81843f442
Step 3/20 : RUN curl -sL https://dl.yarnpkg.com/rpm/yarn.repo -o /etc/yum.repos.d/yarn.repo &&     dnf install --setopt install_weak_deps=false -y     ImageMagick iproute jemalloc less libcurl libcurl-devel     libffi-devel libxml2-devel libxslt-devel nodejs pcre-devel     postgresql postgresql-devel tzdata yarn &&     dnf -y clean all &&     rm -rf /var/cache/yum
 ---> Running in 0f775747949c
The command '/bin/sh -c curl -sL https://dl.yarnpkg.com/rpm/yarn.repo -o /etc/yum.repos.d/yarn.repo &&     dnf install --setopt install_weak_deps=false -y     ImageMagick iproute jemalloc less libcurl libcurl-devel     libffi-devel libxml2-devel libxslt-devel nodejs pcre-devel     postgresql postgresql-devel tzdata yarn &&     dnf -y clean all &&     rm -rf /var/cache/yum' returned a non-zero code: 6
```

Reply:
Buildkite (possibly because of the docker pull commands in the build_containers.sh script) is still failing - now failing early in running dnf (rather than later trying to run yarn):

```
Step 1/20 : FROM quay.io/forem/ruby:3.0.2-testing as builder
3.0.2-testing: Pulling from forem/ruby
4545346f2a49: Pull complete
838a9b65ba57: Pull complete
Digest: sha256:522930f5dd280c0b7aba5c9eb0ba19f3fed53a59822ddceed5d303ea64c08c86
Status: Downloaded newer image for quay.io/forem/ruby:3.0.2-testing
 ---> 99593d0a8366
Step 2/20 : USER root
 ---> Running in d9ebaee0052b
Removing intermediate container d9ebaee0052b
 ---> e9f81843f442
Step 3/20 : RUN curl -sL https://dl.yarnpkg.com/rpm/yarn.repo -o /etc/yum.repos.d/yarn.repo &&     dnf install --setopt install_weak_deps=false -y     ImageMagick iproute jemalloc less libcurl libcurl-devel     libffi-devel libxml2-devel libxslt-devel nodejs pcre-devel     postgresql postgresql-devel tzdata yarn &&     dnf -y clean all &&     rm -rf /var/cache/yum
 ---> Running in 0f775747949c
The command '/bin/sh -c curl -sL https://dl.yarnpkg.com/rpm/yarn.repo -o /etc/yum.repos.d/yarn.repo &&     dnf install --setopt install_weak_deps=false -y     ImageMagick iproute jemalloc less libcurl libcurl-devel     libffi-devel libxml2-devel libxslt-devel nodejs pcre-devel     postgresql postgresql-devel tzdata yarn &&     dnf -y clean all &&     rm -rf /var/cache/yum' returned a non-zero code: 6
```